gsdk/src/storage.rs

```diff
@@ -317,13 +317,18 @@ impl Api {
     ) -> Result<GearPages> {
         let mut pages = HashMap::new();
 
-        for page in &program.pages_with_data {
+        for page in program
+            .pages_with_data
+            .inner
+            .iter()
+            .flat_map(|(s, e)| s.0..=e.0)
```

Comment:
It's not quite clear for me what this 'flat_map' actually does. Perhaps improving the naming (such as 'start' and 'end') or extracting it into a function (as it seems 'flat_map(|(s, e)| s.0..=e.0)' is used in several places) would help.

Reply:
renamed here https://github.com/gear-tech/gear/pull/3791/commits/7002ef4b4a95b6a5198451d80763bd8db4480526

we use here autogenerated types, so I don't think we need to implement for them such functionality